### PR TITLE
Add DeprecationSwitchEvent

### LINKED ISSUE #787: Services must handle `DeprecationSwitch` event
## Problem Definition

On deprecated networks the `DeprecationSwitch` event is emitted. Currently the services cannot handle that and crash

```
2020-05-27T09:12:28.788286576Z Traceback (most recent call last):
2020-05-27T09:12:28.788296129Z   File "/usr/local/lib/python3.7/runpy.py", line 193, in _run_module_as_main
2020-05-27T09:12:28.788441962Z     "__main__", mod_spec)
2020-05-27T09:12:28.788448126Z   File "/usr/local/lib/python3.7/runpy.py", line 85, in _run_code
2020-05-27T09:12:28.788522551Z     exec(code, run_globals)
2020-05-27T09:12:28.788552829Z   File "/services/src/pathfinding_service/cli.py", line 156, in <module>
2020-05-27T09:12:28.788633957Z     main(auto_envvar_prefix="PFS")  # pragma: no cover
2020-05-27T09:12:28.788642217Z   File "/usr/local/lib/python3.7/site-packages/click/core.py", line 764, in __call__
2020-05-27T09:12:28.788839149Z     return self.main(*args, **kwargs)
2020-05-27T09:12:28.788878493Z   File "/usr/local/lib/python3.7/site-packages/click/core.py", line 717, in main
2020-05-27T09:12:28.789028846Z     rv = self.invoke(ctx)
2020-05-27T09:12:28.789034899Z   File "/usr/local/lib/python3.7/site-packages/click/core.py", line 956, in invoke
2020-05-27T09:12:28.789252857Z     return ctx.invoke(self.callback, **ctx.params)
2020-05-27T09:12:28.789288990Z   File "/usr/local/lib/python3.7/site-packages/click/core.py", line 555, in invoke
2020-05-27T09:12:28.789424077Z     return callback(*args, **kwargs)
2020-05-27T09:12:28.789431464Z   File "/services/src/raiden_libs/cli.py", line 243, in call_with_blockchain_info
2020-05-27T09:12:28.789520756Z     return callback(**params)
2020-05-27T09:12:28.789550746Z   File "/services/src/raiden_libs/cli.py", line 168, in call_with_common_options_initialized
2020-05-27T09:12:28.789610094Z     return func(**params)
2020-05-27T09:12:28.789638285Z   File "/services/src/pathfinding_service/cli.py", line 141, in main
2020-05-27T09:12:28.789685341Z     service.get()
2020-05-27T09:12:28.789722125Z   File "src/gevent/greenlet.py", line 795, in gevent._greenlet.Greenlet.get
2020-05-27T09:12:28.789796049Z   File "src/gevent/greenlet.py", line 364, in gevent._greenlet.Greenlet._raise_exception
2020-05-27T09:12:28.789823999Z   File "/usr/local/lib/python3.7/site-packages/gevent/_compat.py", line 62, in reraise
2020-05-27T09:12:28.789905968Z     raise value.with_traceback(tb)
2020-05-27T09:12:28.789933817Z   File "src/gevent/greenlet.py", line 854, in gevent._greenlet.Greenlet.run
2020-05-27T09:12:28.789965825Z   File "/services/src/pathfinding_service/service.py", line 151, in _run
2020-05-27T09:12:28.790032211Z     self._process_new_blocks(to_block)
2020-05-27T09:12:28.790061232Z   File "/services/src/pathfinding_service/service.py", line 176, in _process_new_blocks
2020-05-27T09:12:28.790122179Z     to_block=to_block,
2020-05-27T09:12:28.790149987Z   File "/services/src/raiden_libs/blockchain.py", line 204, in get_blockchain_events
2020-05-27T09:12:28.790217576Z     event = parse_token_network_event(event_dict)
2020-05-27T09:12:28.790245183Z   File "/services/src/raiden_libs/blockchain.py", line 122, in parse_token_network_event
2020-05-27T09:12:28.790294174Z     channel_identifier=event["args"]["channel_identifier"],
2020-05-27T09:12:28.790323836Z   File "/usr/local/lib/python3.7/site-packages/web3/datastructures.py", line 50, in __getitem__
2020-05-27T09:12:28.790384691Z     return self.__dict__[key]  # type: ignore
2020-05-27T09:12:28.790417842Z KeyError: 'channel_identifier'
```

```
(Pdb) event
AttributeDict({'args': AttributeDict({'new_value': True}), 'event': 'DeprecationSwitch', 'logIndex': 123, 'transactionIndex': 110, 'transactionHash': HexBytes('0xc3a92660d806e47e7fa003caf5f45dc84933e5e729a46f7bbece196ceeb90fc9'), 'address': '0x4261f6Eaf3aCf568a5EE6E15bE0DFCDA379dAc8B', 'blockHash': HexBytes('0x495bc4f4f3c46ef458fe683c238bb13e1bacbbb2ba080878d5520176f875cb2e'), 'blockNumber': 10134001})
```

## Actual Outcome

Services crash

## Expected Outcome

Services continue to work

### PR DESCRIPTION
**Pull request by palango**
Fixes #787

### CODE REVIEW

File: src/raiden_libs/blockchain.py [hunk 1]
```diff
@@ -119,30 +120,40 @@ def parse_token_network_event(event: dict) -> Optional[Event]:
...
-        return ReceiveChannelSettledEvent(**common_infos)
+        return ReceiveChannelSettledEvent(
+            channel_identifier=event["args"]["channel_identifier"],
+            **common_infos,
+        )
+
+    # FIXME: use value from ChannelEvent as soon as PR is merged
+    # https://github.com/raiden-network/raiden-contracts/pull/1389
+    if event_name == "DeprecationSwitch":
+        return DeprecationSwitchEvent(**common_infos)
```
Comment (palango) [inline src/raiden_libs/blockchain.py:156]: We can even argue if we should add the event as long as we don't use it to trigger a certain action.

File: src/raiden_libs/blockchain.py [hunk 2]
```diff
@@ -118,31 +119,39 @@ def parse_token_network_event(event: dict) -> Optional[Event]:
     event_name = event["event"]
 
     common_infos = dict(
-        token_network_address=decode_hex(event["address"]),
-        channel_identifier=event["args"]["channel_identifier"],
-        block_number=event["blockNumber"],
+        token_network_address=decode_hex(event["address"]), block_number=event["blockNumber"],
     )
```
Comment (karlb) [inline src/raiden_libs/blockchain.py:123]: We could leave the common_infos unchanged if we move the `if event_name == "DeprecationSwitch"` before this. I don't see a downside and it keeps the code more compact.

Comment (palango) [reply src/raiden_libs/blockchain.py:123]: What do you think about the event itself, after sleeping over it I don't see any actual use, so I'd remove it.

Comment (karlb) [reply src/raiden_libs/blockchain.py:123]: Yes, probably a good idea, too. We don't need to do anything on that event.

**Pull request merged**